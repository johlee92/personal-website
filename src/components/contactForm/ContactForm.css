.contact-form {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    white-space: pre-line;
}

.contact-form form {
    margin: 10px 0;
    width: 80%;
    max-width: 720px;
}

.form-group {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 10px;
}

.form-control {
    min-width: 150px;
    width: 70%;
}