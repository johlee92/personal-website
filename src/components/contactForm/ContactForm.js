import React, { Component } from 'react';
import './ContactForm.css';

class ContactForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            name: '',
            email: '',
            message: ''
        }
    }

    onNameChange = (event) => {
        this.setState({name: event.target.value})
    }
    
    onEmailChange = (event) => {
        this.setState({email: event.target.value})
    }
    
    onMessageChange = (event) => {
        this.setState({message: event.target.value})
    }
    
    handleSubmit = (event) => {
        event.preventDefault();
        console.log(this.state);
    }

    render() {
        return (
            <div className="contact-form">
                <form id="contact-form" onSubmit={this.handleSubmit} method="POST">
                    <h2>Contact Form</h2>
                    <div className="form-group">
                        <label htmlFor="name">Name</label>
                        <input type="text" className="form-control" />
                    </div>
                    <div className="form-group">
                        <label htmlFor="exampleInputEmail1">Email address</label>
                        <input type="email" className="form-control" aria-describedby="emailHelp" />
                    </div>
                    <div className="form-group">
                        <label htmlFor="message">Message</label>
                        <textarea className="form-control form-message" rows="5"></textarea>
                    </div>
                    <button type="submit" className="projectCard">Submit</button>
                </form>
            </div>
        )
    }
}

export default ContactForm;