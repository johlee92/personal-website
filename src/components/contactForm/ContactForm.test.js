import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ContactForm from './ContactForm';

describe(`ContactForm component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
                <ContactForm />
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component as expected', () => {
        const tree = renderer
            .create(
                <ContactForm />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})
