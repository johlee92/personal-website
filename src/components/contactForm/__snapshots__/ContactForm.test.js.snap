// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component renders the component as expected 1`] = `
<div
  className="contact-form"
>
  <form
    id="contact-form"
    method="POST"
    onSubmit={[Function]}
  >
    <h2>
      Contact Form
    </h2>
    <div
      className="form-group"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="exampleInputEmail1"
      >
        Email address
      </label>
      <input
        aria-describedby="emailHelp"
        className="form-control"
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        className="form-control form-message"
        rows="5"
      />
    </div>
    <button
      className="projectCard"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
